api/cmd/conformance-tests/main.go

```
@@ -0,0 +1,194 @@
+	}
+
+	kubermaticClient := kubermaticclientset.NewForConfigOrDie(config)
+
+	if opts.cleanupOnStart {
+		if opts.namePrefix == "" {
+			log.Fatalf("cleanup-on-start was specified but name-prefix is empty")
+		}
+
+		clusterList, err := kubermaticClient.KubermaticV1().Clusters().List(metav1.ListOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		for _, cluster := range clusterList.Items {
+			if strings.HasPrefix(cluster.Name, opts.namePrefix) {
+				p := metav1.DeletePropagationBackground
+				opts := metav1.DeleteOptions{PropagationPolicy: &p}
+				fmt.Println("Deleting cluster " + cluster.Name + "...")
+				if err = kubermaticClient.KubermaticV1().Clusters().Delete(cluster.Name, &opts); err != nil {
+					log.Fatalf("failed to delete cluster %s: %v", cluster.Name, err)
+				}
+			}
+		}
+
+		fmt.Println("Cleaned up all old clusters")
+		os.Exit(0)
+	}
+
+	kubeClient := kubernetes.NewForConfigOrDie(config)
+	kubermaticInformerFactory := kubermaticinformers.NewSharedInformerFactory(kubermaticClient, informer.DefaultInformerResyncPeriod)
+	kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, informer.DefaultInformerResyncPeriod)
+
+	opts.kubermaticClient = kubermaticClient
+	opts.clusterLister = kubermaticInformerFactory.Kubermatic().V1().Clusters().Lister()
+
+	opts.clusterClientProvider = clusterclient.New(kubeInformerFactory.Core().V1().Secrets().Lister())
+
+	kubermaticInformerFactory.Start(rootCtx.Done())
+	kubeInformerFactory.Start(rootCtx.Done())
+	kubermaticInformerFactory.WaitForCacheSync(rootCtx.Done())
+	kubeInformerFactory.WaitForCacheSync(rootCtx.Done())
+
+	glog.Info("Starting E2E tests...")
+
+	var scenarios []testScenario
+	scenarios = append(scenarios, getAWSScenarios()...)
+	scenarios = append(scenarios, getDigitaloceanScenarios()...)
+	scenarios = append(scenarios, getHetznerScenarios()...)
+	scenarios = append(scenarios, getOpenStackScenarios()...)
+
+	runner := newRunner(scenarios, &opts)
+	hadFailure, err := runner.Run()
```

Why not return an error in case of failure?